pkg/asset/agent/image/agentartifacts.go

```diff
@@ -175,3 +190,26 @@ func (a *AgentArtifacts) Files() []*asset.File {
 	// Return empty array because File will never be loaded.
 	return []*asset.File{}
 }
+
+func extractRootFS(bootArtifactsFullPath, agentISOPath, arch string) error {
+	os.RemoveAll(bootArtifactsFullPath)
+
+	err := os.Mkdir(bootArtifactsFullPath, 0750)
```

Comment:
The creation the `bootArtifactsFullPath` dir hidden within this method is confusing. Especially for the PXE assets, it's used to store other artifacts. At the cost of a minimal duplication, I'd find easier to read such task in the main assets code